### PR TITLE
Implemented the structure of smos-report

### PR DESCRIPTION
**Pull request by CSVdB**

### CODE REVIEW

File: smos-report/src/Smos/Report/OptParse.hs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    config <- getConfig flags
+    (getDispatch cmd, ) <$> getSettings flags config
+
+getConfig :: Flags -> IO Configuration
+getConfig Flags {..} = do
+    cfgPath <- fromMaybe defaultConfigFile $ resolveFile' <$> flagsConfigFile
+    config <- load [Optional $ toFilePath cfgPath]
+    Configuration <$> lookup config "workDir" <*> lookup config "shouldPrint"
+  where
+    defaultConfigFile = (</>) <$> getHomeDir <*> parseRelFile ".wfrc"
```
Comment (NorfairKing) [inline smos-report/src/Smos/Report/OptParse.hs:29]: This doesn't look right.

File: smos-report/package.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+name: smos-report
+version: '0.0.0.0'
+category: Smos
+author: Nick Van den Broeck
+maintainer: syd@cs-syd.eu
+copyright: ! 'Copyright: (c) 2018 Tom Sydney Kerckhove'
+license: AllRightsReserved
+homepage: https://github.com/NorfairKing/smos
+
+executables:
```
Comment (NorfairKing) [inline smos-report/package.yaml]: Why is the executable section before the library section, that is atypical

File: smos-report/src/Smos/Report/OptParse/Types.hs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    } deriving (Show, Eq)
+
+data Configuration = Configuration
+    { cfgWorkDir :: Maybe FilePath
+    , cfgShouldPrint :: Maybe ShouldPrint
+    } deriving (Show, Eq)
+
+data Settings = Settings
+    { sttWorkDir :: Path Abs Dir
+    , sttShouldPrint :: ShouldPrint
```
Comment (NorfairKing) [inline smos-report/src/Smos/Report/OptParse/Types.hs]: This `stt` and `cfg` prefix are also atypical.
Just `set` and `config` usually.

File: smos-report/package.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - base
+    - smos-report
+
+library:
+  source-dirs: src/
+  ghc-options: -Wall
+  default-extensions: NoImplicitPrelude
+  dependencies:
+  - base
+  - configurator
```
Comment (NorfairKing) [inline smos-report/package.yaml:16]: I'm not convinced that `configurator` is the right tool for the job.
`yaml` is probably simpler.

File: smos-report/src/Smos/Report/OptParse/Types.hs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    = PrintError
+    | PrintWarning
+    | DontPrint
+    deriving (Show, Eq)
+
+parseShouldPrint :: String -> Maybe ShouldPrint
+parseShouldPrint "error" = Just PrintError
+parseShouldPrint "warning" = Just PrintWarning
+parseShouldPrint "nothing" = Just DontPrint
+parseShouldPrint _ = Nothing
```
Comment (NorfairKing) [inline smos-report/src/Smos/Report/OptParse/Types.hs]: These two lines are a bit confusing.
I would imagine that `nothing` would be parsed as `Nothing`.
Not a big deal.

File: smos-report/src/Smos/Report/OptParse/Types.hs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+data Command
+    = CommandWaiting
+    | CommandNext
+    deriving (Show, Eq)
+
+data Flags = Flags
+    { flagsConfigFile :: Maybe FilePath
+    , flagsWorkDir :: Maybe FilePath
+    , flagsShouldPrint :: Maybe ShouldPrint
```
Comment (NorfairKing) [inline smos-report/src/Smos/Report/OptParse/Types.hs]: The prefix here is usually `flag`.
Not a big deal.

File: smos-report/package.yaml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    - -rtsopts
+    - -with-rtsopts=-N
+    dependencies:
+    - base
+    - smos-report
+
+library:
+  source-dirs: src/
+  ghc-options: -Wall
+  default-extensions: NoImplicitPrelude
```
Comment (NorfairKing) [inline smos-report/package.yaml]: This messes with using `ghci`.
I'd rather have all the extensions IN the file.

**Review by NorfairKing — CHANGES_REQUESTED** [review]
Mostly nits.

File: smos-report/src/Smos/Report/OptParse.hs (hunk 1 above)
Comment (CSVdB) [reply smos-report/src/Smos/Report/OptParse.hs:29]: Is the problem the name "wfrc"?

File: smos-report/package.yaml (hunk 4 above)
Comment (CSVdB) [reply smos-report/package.yaml:16]: Configurator is very easy to use, both in the Haskell code and for the configuration file, and config files is exactly what configurator is meant for.
Why do you believe yaml is simpler?

**Comment by NorfairKing** [comment]
Please rebase, because the build fails. Also please add smos-report to the nix build so that it's in the release and travis will build it too.

**Pull request merged**